apis/projectcontour/v1/httpproxy.go

```diff
@@ -101,17 +101,120 @@ type HeaderMatchCondition struct {
 	NotExact string `json:"notexact,omitempty"`
 }
 
+// ExtensionServiceReference names an ExtensionService resource.
+type ExtensionServiceReference struct {
+	// API version of the referent.
+	// If this field is not specified, the default "projectcontour.io/v1alpha1" will be used
+	//
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,5,opt,name=apiVersion"`
+
+	// Kind of the referent.
+	// If this field is not specified, the default "ExtensionService" will be used.
```

Comment:
If we supported a different resource types, then Contour would be watching that.

Reply:
This is the standard way to do a cross-type reference now, so I think we should stick with the overall Kubernetes API conventions.

Reply:
It does make it easier to do the migration for people from v1alpha1 to v1 when we do that, since you can request a specific version. I could see removing the Kind, I suppose, and we can add it later if you want?

Reply:
In summary; I'm removing Kind.